assets/dev/js/editor/document/elements/commands/paste-style.js

```diff
@@ -96,26 +96,55 @@ export class PasteStyle extends CommandHistory {
 					diffSettings[ controlName ] = controlSourceValue;
 				} );
 
-				// Moved from `editor/elements/views/base.js` `pasteStyle` function.
-				targetContainer.view.allowRender = false;
+				this.pasteStyle( targetContainer, diffSettings );
+			} );
+		} );
+	}
 
-				// BC: Deprecated since 2.8.0 - use `$e.hooks`.
-				elementor.channels.data.trigger( 'element:before:paste:style', targetContainer.model );
+	/**
+	 * @param {Container} targetContainer
+	 * @param {{}} settings
+	 */
+	pasteStyle( targetContainer, settings ) {
+		targetContainer.view.allowRender = false;
 
-				$e.run( 'document/elements/settings', {
-					container: targetContainer,
-					settings: diffSettings,
-					options: { external: true },
-				} );
+		// BC: Deprecated since 2.8.0 - use `$e.hooks`.
+		elementor.channels.data.trigger( 'element:before:paste:style', targetContainer.model );
 
-				// BC: Deprecated since 2.8.0 - use `$e.hooks`.
-				elementor.channels.data.trigger( 'element:after:paste:style', targetContainer.model );
+		const globals = settings.__globals__;
 
-				targetContainer.view.allowRender = true;
+		if ( globals ) {
+			delete settings.__globals__;
+		}
 
-				targetContainer.render();
-			} );
+		$e.run( 'document/elements/settings', {
+			container: targetContainer,
+			settings: settings,
+			options: {
+				external: true,
+				render: false,
+			},
 		} );
+
+		if ( globals ) {
+			$e.run( 'document/globals/settings', {
+				container: targetContainer,
+				settings: globals,
+				options: {
+					external: true,
+					render: false,
+				},
+			} );
+
+			targetContainer.panel.refresh();
+		}
+
+		// BC: Deprecated since 2.8.0 - use `$e.hooks`.
+		elementor.channels.data.trigger( 'element:after:paste:style', targetContainer.model );
+
+		targetContainer.view.allowRender = true;
```

Comment:
Remove, the settings has `render: false`.

Reply:
Handled.